contracts/core/Dispatcher.sol

```diff
@@ -810,13 +802,16 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, ReentrancyGuard, IDi
      * @return lightClient The light client associated with the connection.
      */
     function _getLightClientFromConnection(string memory connection) internal view returns (ILightClient lightClient) {
+        if (bytes(connection).length == 0) {
```

Comment:
is this not checked before? Consider applying the same rules as above (ie connection.length >= 2)

Reply:
connection is a string (not an array of strings) - so this line checks that the strings stored in connection hops are nonempty strings, whereas what you are proposing checks the length of the string array itself (not relevant here?)

Reply:
this validation should be added now!